### PR TITLE
Add propelauth provider

### PR DESCRIPTION
**Pull request by pfvatterott**
Add [PropelAuth](https://www.propelauth.com/) provider

### CODE REVIEW

File: src/PropelAuth/Provider.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+     */
+    protected function getPropelAuthUrl()
+    {
+        $baseUrl = $this->getConfig('auth_url');
+
+        if ($baseUrl === null) {
+            throw new InvalidArgumentException('Missing Auth URL value.');
+        }
+
+        return $baseUrl;
```
Comment (lucasmichot) [inline src/PropelAuth/Provider.php:33]:
```suggestion
        return rtrim($baseUrl, '/');
```

File: src/PropelAuth/Provider.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+        return json_decode((string) $response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User)->setRaw($user)->map($user);
```
[the remark this answers is not in the document]
Comment (pfvatterott) [reply src/PropelAuth/Provider.php:71]: Would this method not work? We are an auth provider (similar to Auth0 or Clerk) and allow our customers to customize user fields. So for example, one of our customers may have a custom field called "favoriteSport" which is returned in the OAuth flow. If not, what would you recommend here?

Really appreciate the help :)

**Comment by pfvatterott** [comment]
@lucasmichot done! Thank you :)

**Comment by lucasmichot** [comment]
Thanks @pfvatterott , can you also make a PR against https://github.com/SocialiteProviders/website ?

**Review by lucasmichot — APPROVED** [review]
GTM @atymic

**Comment by pfvatterott** [comment]
> Thanks @pfvatterott , can you also make a PR against https://github.com/SocialiteProviders/website ?

[Done](https://github.com/SocialiteProviders/website/pull/71), thank you! @lucasmichot

**Review by atymic — APPROVED** [review]

**Comment by pfvatterott** [comment]
Hey @lucasmichot! It looks like this has not been published on packagist. I tried adding it myself but it appears I cannot since I'm not a SocialiteProviders maintainer. Is this something you can do for me? Or if possible, could you instruct me on what I should do here?

Thanks!

**Comment by pfvatterott** [comment]
@lucasmichot @atymic What do I need to do to get this published? It's currently not on Packagist

**Comment by atymic** [comment]
It's been tagged now :)

**Comment by pfvatterott** [comment]
Thank you @atymic ! :)

**Pull request merged**